src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperRunner.cs

```diff
@@ -24,7 +24,10 @@ public class TagHelperRunner
                 executionContext.AllAttributes,
                 executionContext.UniqueId,
                 executionContext.GetChildContentAsync);
-            var tagHelperOutput = new TagHelperOutput(executionContext.TagName, executionContext.HTMLAttributes);
+            var tagHelperOutput = new TagHelperOutput(executionContext.TagName, executionContext.HTMLAttributes)
+            {
+                SelfClosing = executionContext.SelfClosing,
```

Comment:
I disagree with making this a a constructor required property for the `TagHelperOutput`, the `TagHelperOutput` is a living object which can be modified by users and therefore `SelfClosing` is malleable. That being said we did decide not to touch the property for our own `TagHelper`s in MVC; doesn't mean other `TagHelper` authors will not though. I can even see people constructing their own `TagHelperOutput`s to build complex structures; I'd hate to require them to provide if it's self closing on construction.